### PR TITLE
[no ticket][risk=no] Improve circleci e2e job configuration

### PR DESCRIPTION
**Pull request by aweng98**
Hoping to reduce e2e job total run time.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -319,6 +321,14 @@ commands:
           # Stop running Puppeteer end-to-end test if code in relevant directories is unchanged.
           dir_names: "ui e2e .circleci"
 
+  puppeteer-access-test-user-setup:
```
Comment (aweng98) [inline .circleci/config.yml:324]: Making `puppeteer-access-test-user-setup` as a CI command instead a CI job. Easier to call in other jobs.

File: .circleci/config.yml [hunk 2]
```diff
@@ -602,38 +611,20 @@ jobs:
...
-    executor: workbench-executor
-    steps:
-      - checkout-code
-      - attach_workspace:
-          at: .
-      - gcloud-auth-login
-      - run:
-          working_directory: ~/workbench/api
-          name: Prep the e2e access test user (in Test env) by setting access module timestamps
-          command: ./project.rb set-access-module-timestamps --user ${PUPPETEER_ACCESS_TEST}
```
Comment (aweng98) [inline .circleci/config.yml:636]: This was the job that converted to command.

File: .circleci/config.yml [hunk 3]
```diff
@@ -537,23 +547,22 @@ jobs:
...
-            echo "WORKBENCH_ENV=${WORKBENCH_ENV}"
-            echo "TEST_MODE=${TEST_MODE}"
-      - run:
-          name: "Export user password"
           command: |
+            cp .env.circleci.<< parameters.env_name >> .env
             echo "export PASSWORD=$PUPPETEER_USER_PASSWORD" >> "$BASH_ENV"
             source $BASH_ENV
+            echo "WORKBENCH_ENV=${WORKBENCH_ENV}"
+            echo "TEST_MODE=${TEST_MODE}"
```
Comment (aweng98) [inline .circleci/config.yml:565]: Grouped three `run` steps into one step for easier read.

**Comment by aweng98** [comment]
@calbach Hi, just want to to keep you informed about this PR. I plan to use `!pr` to find a reviewer tomorrow.

**Review by freemabd — APPROVED** [review]

**Pull request merged**